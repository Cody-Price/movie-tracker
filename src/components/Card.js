import React, { Component } from 'react';
import './Card.scss'
import {connect} from 'react-redux';
import { postFavorite, deleteFavorite } from '../API';
import { addFavorite, deleteFavoriteFromStore } from '../actions';

class Card extends Component {
  constructor(props) {
    super(props)
  }
  
  toggleFavorite = () => {
    const {movie, user, favorites} = this.props;
    const matchingFavorite = favorites.find(faveMovie => {
      return faveMovie.movie_id === movie.movie_id
    })
    
    if(!matchingFavorite) {
      postFavorite(movie, user.id);
      this.props.addFavorite(movie);
    } else {
      deleteFavorite(user.id, movie.movie_id);
      this.props.deleteFavoriteFromStore(movie)
    }
  }
  
  render() {

    const faves = this.props.favorites.find(fave => {
      return fave.movie_id === this.props.movie.movie_id
    })
    let favoriteClass = faves ? 'favorite-movie favorite-btn' : 'favorite-btn';

    let loggedIn = this.props.user.name ? false : true;
    const posterPath = `https://image.tmdb.org/t/p/w500/${this.props.movie.poster_path}`
    return (
      <div className="movie-card">
        <button className={favoriteClass}
          disabled={loggedIn} 
          onClick={this.toggleFavorite}
          data-tooltip={loggedIn ? 'Must be signed in to favorite movies' : null}
        >
            Favorite
        </button>
        <article className="card-text">
          <h1 className="movie-title">{this.props.movie.title}</h1>
          <p> {this.props.movie.overview} </p>
        </article>
        <article className='poster'>
          <img src={posterPath} />
        </article>
      </div>
    )
  }
}

const mapStateToProps = (state) => ({
  user: state.user,
  movies: state.movies,
  favorites: state.favorites
})

const mapDispatchToProps = (dispatch) => ({
  addFavorite: (favorite) => dispatch(addFavorite(favorite)),
  deleteFavoriteFromStore: (favorite) => dispatch(deleteFavoriteFromStore(favorite))
})

export default connect(mapStateToProps, mapDispatchToProps)(Card);