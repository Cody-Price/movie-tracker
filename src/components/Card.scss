@import url('https://fonts.googleapis.com/css?family=Acme');
$acme-font: 'Acme', sans-serif;

.card-text {
  opacity: 0;
  position: absolute;
  z-index:  4;
}

.movie-card {
  position: relative;
  border-radius: 2px;
  box-shadow: 4px 6px 6px rgba(255, 255, 255, 0.6);
  color: white;
  font-family: $acme-font;
  height: 450px;
  margin: 20px;  
  overflow: hidden;
  transition-duration: 0.4s;
  width: 300px;
  &:hover {
    transform: scale(1.1);
    transition-duration: 0.4s;
    cursor: pointer;
    img {
      filter: blur(16px); 
    }
    .card-text {
      width: 80%;
      transition: ease-in .4s;
      transition-delay: .3s;
      height: 80%;
      padding: 3px;
      opacity: 1;
      display: flex;
      flex-direction: column;
      justify-content: space-between;
      align-items: center;
      position: absolute;
      /*left: 0;*/
      margin: 16px;
      font-size: 16px;
      text-align: center;
    }
  }
}

.movie-title {
  font-size: 36px;
  -webkit-text-stroke-width: .1px;
  -webkit-text-stroke-color: black;
}

.poster {
  img {
    width: 100%;
  }
}


.favorite-btn {
  position: absolute;
  z-index: 5;
  left: 200px;
  top: 9px;

  &:disabled {
    background-color: grey;
  }

  &:hover:disabled {
    transform: scale(1);
  }

}