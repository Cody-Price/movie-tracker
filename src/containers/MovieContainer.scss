
.movie-container {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  padding: 16px;
  background: black;
  margin-top: 150px;
  z-index: 1;
}