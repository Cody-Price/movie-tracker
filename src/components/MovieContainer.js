import React, { Component } from 'react'
import Card from './Card'


const MovieContainer = (props) => {
  const movies = props.movies.map(movie => {
    return <Card movie={movie}/>
  })

  return (
    <div>
      { movies }
    </div>
  )
}

export default MovieContainer;