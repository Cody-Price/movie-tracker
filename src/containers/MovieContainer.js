import React from 'react'
import Card from '../components/Card'
import { connect } from 'react-redux'
import './MovieContainer.scss'
import SearchBar from './SearchBar'

export const MovieContainer = (props) => {
	let displayedMovies
	if (props.match.path === '/favorites') {
		displayedMovies = props.favorites
	} else {
		displayedMovies = props.movies
	}
	const movies = displayedMovies.map(movie => {
		return <Card movie={movie} key={movie.movie_id}/>
	})

	if (props.isLoading) {
		return (
			<div className="loading">
        LOADING...
			</div>
		)
	} else {
		return (
			<div>
				<SearchBar />
				<div className={displayedMovies.length === 0 ? 'hidden' : 'movie-container'}>
					{ movies }
				</div>
				<h1 className={displayedMovies.length === 0 && props.match.path === '/favorites' ? 'no-movies movie-container' : 'hidden'}>
          You have no favorited movies silly!
				</h1>
				<h1 className={displayedMovies.length === 0 && props.match.path === '/' ? 'no-movies movie-container' : 'hidden'}>
          No movies matched your search query.
				</h1>
			</div>
		)
	}
}

export const mapStateToProps = (state) => ({
	movies: state.movies,
	favorites: state.favorites,
	isLoading: state.isLoading,
	error: state.error
})

export default connect(mapStateToProps)(MovieContainer)