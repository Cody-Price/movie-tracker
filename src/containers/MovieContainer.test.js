import React from 'react';
import { shallow } from 'enzyme';
import MovieContainer from './MovieContainer'
import { mapStateToProps } from './MovieContainer'
import Card from '../components/Card'
import { connect } from 'react-redux'


describe('MovieContainer', () => {
  let wrapper
  let mockFavorites
  let mockMovies

  beforeEach(() => {
    mockMovies = [{title: 'movie1'}, {title: 'movie2'},{title: 'movie3'}]
    mockFavorites = [{title: 'movie1'}, {title: 'movie2'}]
    wrapper = shallow(
      <MovieContainer
        favorites={mockFavorites}
        movies={mockMovies}
        match={{path: '/favorites'}}
      />
    )
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('mapStateToProps', () => {
    it('should return an object with a movies and favorites array', () => {
      const mockState = {
        movies: mockMovies,
        favorites: mockFavorites
      }
      const expected = { movies: mockState.movies, favorites: mockState.favorites }
      const result = mapStateToProps(mockState)
      expect(result).toEqual(expected)
    })
  })
})