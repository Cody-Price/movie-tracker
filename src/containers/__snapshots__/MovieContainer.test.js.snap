// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieContainer should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MovieContainer
    favorites={
      Array [
        Object {
          "movie_id": 1,
          "title": "movie1",
        },
        Object {
          "movie_id": 2,
          "title": "movie2",
        },
      ]
    }
    match={
      Object {
        "path": "/",
      }
    }
    movies={
      Array [
        Object {
          "movie_id": 1,
          "title": "movie1",
        },
        Object {
          "movie_id": 2,
          "title": "movie2",
        },
        Object {
          "movie_id": 3,
          "title": "movie3",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="movie-container"
        >
          <Connect(SearchBar) />
          <Connect(Card)
            movie={
              Object {
                "movie_id": 1,
                "title": "movie1",
              }
            }
          />
          <Connect(Card)
            movie={
              Object {
                "movie_id": 2,
                "title": "movie2",
              }
            }
          />
          <Connect(Card)
            movie={
              Object {
                "movie_id": 3,
                "title": "movie3",
              }
            }
          />
        </div>,
        <h1
          className="hidden"
        >
          You have no favorited movies silly!
        </h1>,
        <h1
          className="hidden"
        >
          No movies matched your search query.
        </h1>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Connect(SearchBar) />,
            Array [
              <Connect(Card)
                movie={
                  Object {
                    "movie_id": 1,
                    "title": "movie1",
                  }
                }
              />,
              <Connect(Card)
                movie={
                  Object {
                    "movie_id": 2,
                    "title": "movie2",
                  }
                }
              />,
              <Connect(Card)
                movie={
                  Object {
                    "movie_id": 3,
                    "title": "movie3",
                  }
                }
              />,
            ],
          ],
          "className": "movie-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "movie": Object {
                "movie_id": 1,
                "title": "movie1",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "class",
            "props": Object {
              "movie": Object {
                "movie_id": 2,
                "title": "movie2",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "class",
            "props": Object {
              "movie": Object {
                "movie_id": 3,
                "title": "movie3",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "You have no favorited movies silly!",
          "className": "hidden",
        },
        "ref": null,
        "rendered": "You have no favorited movies silly!",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "No movies matched your search query.",
          "className": "hidden",
        },
        "ref": null,
        "rendered": "No movies matched your search query.",
        "type": "h1",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="movie-container"
          >
            <Connect(SearchBar) />
            <Connect(Card)
              movie={
                Object {
                  "movie_id": 1,
                  "title": "movie1",
                }
              }
            />
            <Connect(Card)
              movie={
                Object {
                  "movie_id": 2,
                  "title": "movie2",
                }
              }
            />
            <Connect(Card)
              movie={
                Object {
                  "movie_id": 3,
                  "title": "movie3",
                }
              }
            />
          </div>,
          <h1
            className="hidden"
          >
            You have no favorited movies silly!
          </h1>,
          <h1
            className="hidden"
          >
            No movies matched your search query.
          </h1>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Connect(SearchBar) />,
              Array [
                <Connect(Card)
                  movie={
                    Object {
                      "movie_id": 1,
                      "title": "movie1",
                    }
                  }
                />,
                <Connect(Card)
                  movie={
                    Object {
                      "movie_id": 2,
                      "title": "movie2",
                    }
                  }
                />,
                <Connect(Card)
                  movie={
                    Object {
                      "movie_id": 3,
                      "title": "movie3",
                    }
                  }
                />,
              ],
            ],
            "className": "movie-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "class",
              "props": Object {
                "movie": Object {
                  "movie_id": 1,
                  "title": "movie1",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "class",
              "props": Object {
                "movie": Object {
                  "movie_id": 2,
                  "title": "movie2",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "3",
              "nodeType": "class",
              "props": Object {
                "movie": Object {
                  "movie_id": 3,
                  "title": "movie3",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "You have no favorited movies silly!",
            "className": "hidden",
          },
          "ref": null,
          "rendered": "You have no favorited movies silly!",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "No movies matched your search query.",
            "className": "hidden",
          },
          "ref": null,
          "rendered": "No movies matched your search query.",
          "type": "h1",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
